### PR TITLE
server: minor changes for add/remove peer.

### PR DESCRIPTION
**Pull request by disksing**
@qiuyesuifeng

### CODE REVIEW

File: server/cluster_worker.go [hunk 1]
```diff
@@ -40,71 +40,37 @@ func (c *raftCluster) handleAddPeerReq(region *metapb.Region) (*metapb.Peer, err
...
-
-	region.Peers = append(region.Peers, peer)
-
-	return peer, nil
+	return nil, errors.Errorf("find no store to add peer for region %v", region)
 }
 
 // If leader is nil, we will return an error, or else we can remove none leader peer.
-func (c *raftCluster) handleRemovePeerReq(region *metapb.Region, leader *metapb.Peer) (*metapb.Peer, error) {
+func (c *raftCluster) handleRemovePeerReq(region *metapb.Region, leaderID uint64) (*metapb.Peer, error) {
```
Comment (qiuyesuifeng) [inline server/cluster_worker.go]: Should we check the validity of leaderID?

Comment (disksing) [inline server/cluster_worker.go]: @qiuyesuifeng I think it's acceptable not to check it. And if we want to, maybe better to check it outside of this func.

**Comment by ngaut** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**